### PR TITLE
206 Show most recent Engagement information in EngagementDetailView for each ContactListItem

### PR DESCRIPTION
**Pull request by will0684**
# Description

In the EngagementsDetailView, all contacts are fetched and that data is used to find the most recent Engagement for each of the contacts that the EngagementDetailView has in its ContactList and display the related information. Child engagements to Contacts data/documents are returned from the db in chronological order so it is a matter of just taking the element from the end of the array and passing it's values to the EngShowContacts component.

## Test Instructions

1. npm run mongodb
1. npm run seed
1. npm run dev
1. navigate to EngagementDetailView by selecting an Engagement in the EngagementList of the EngagementSearch
1. see that contact details for that engagement include details on the latest engagement

## Checklist
- [x] Strings use placeholders for translation (No hard coded strings)
- [ ] Unit tests have been added/updated

### CODE REVIEW

File: pages/view/engagement/_id.vue [hunk 1]
```diff
@@ -77,6 +77,9 @@
         :index="index"
         :name="con.keyContactName"
         :orgname="con.orgName"
+        :last="getLastEngagement(con._id).subject"
+        :date="getLastEngagement(con._id).date"
+        :number="getLastEngagement(con._id).numParticipants"
```
Comment (andr0272) [inline pages/view/engagement/_id.vue:82]: Would it be possible to simplify this? Get the engagement once and populate the fields accordingly?

File: store/contacts.js [hunk 2]
```diff
@@ -36,3 +36,9 @@ export const actions = {
     })
   }
 }
+
+// export const getters = {
+//   getContactByIdandReturnLastEngagement: (state) => (id) => {
+//     return state.contacts.find
+//   }
+// }
```
Comment (andr0272) [inline store/contacts.js:44]: :vomiting_face:

Comment (will0684) [reply store/contacts.js:44]: removing this now 😳

File: pages/view/engagement/_id.vue (hunk 1 above)
Comment (will0684) [reply pages/view/engagement/_id.vue:82]: You are totally correct, very inefficient implementation. I will take another look at it.

Comment (will0684) [reply pages/view/engagement/_id.vue:82]: So I improved the efficiency significantly removing a step of the implementation that would take state and clone it. It wasn't necessary but it was required when I was doing changes to some of the data that are no longer required. As to getting it once and populating the fields, we are populating this data with 2 different arrays, specifically Engagement documents with their related child Contacts and Contacts documents with their related child Engagements documents.

In this page at the first layer we get the Engagement + it's child Contacts data to display the Engagement details and a list of related Contacts. For each one of those Contacts, we need to have their related Engagement information to be able to populate the last engagement field of the list component. Right now, our query returns the first and second layers of data i.e. Engagements or Contacts with their related data so in this case we need to make 2 queries to get the data we need, which is; 1 Engagement with all related Contacts + all those same Contacts with all their Engagement information so we can get the most recent.

[There is a way in Mongoose to get nested nested data in a query](https://stackoverflow.com/a/34444982) and I believe that would be more efficient for the client side but I am not sure if it is worth it in this case. What do you think?

Comment (andr0272) [reply pages/view/engagement/_id.vue:82]: Not worth the effort IMO. Thanks for the explanation. :+1: Approving now.

**Pull request merged**